src/main/java/org/jetbrains/plugins/innerbuilder/InnerBuilderGenerator.java

```diff
@@ -298,7 +298,9 @@ private PsiMethod generateBuilderSetter(final PsiType builderType, final PsiFiel
             methodName = fieldName;
         }
 
-        final String parameterName = !BUILDER_SETTER_DEFAULT_PARAMETER_NAME.equals(fieldName) ?
+        final String parameterName = options.contains(InnerBuilderOption.FIELD_NAMES) ? 
+                fieldNames :
+                !BUILDER_SETTER_DEFAULT_PARAMETER_NAME.equals(fieldName) ?
```

Comment:
I do not think this would work. The generated method would be:

```
public Builder withSomeField(String fieldName) {
    fieldName = fieldName;
    return this;
}
```

There are more changes required for this to work on the current code base. E.g. need to attach `this.`.